scrnaseq.Rmd

```diff
@@ -513,16 +863,16 @@ The first plot is based on the "JackStraw" procedure: parts of the data is repea
 The second plot is an "Elbow plot": PCs are ranked based on the percentage of variance they explain. 
 
 For your dataset, we decided to go for `r param$pc_n` PCs. 
-```{r dimensionality, results="hide"}
-# NOTE: This process can take a long time for big datasets, comment out for expediency. More
-# approximate techniques such as those implemented in ElbowPlot() can be used to reduce
-# computation time
-sc = Seurat::JackStraw(sc, num.replicate=100, dims=20, assay="RNA")
-sc = Seurat::ScoreJackStraw(sc, dims=1:20)
-p = Seurat::JackStrawPlot(sc, dims=1:20)
-p = PlotMystyle(p, title="Jack Straw plot", legend_position="bottom")
-p
+```{r dimensionality}
+if (DefaultAssay(sc)!="SCT" & DefaultAssay(sc)!="SCTintegrated") {
```

Comment:
Mmmh, If I remember correctly, I had assay="RNA" here before. Why change? The default assay can be whatever, we can still use JackStraw?

Reply:
I thought that it did not work with SCT?

Reply:
No. That is why I specifically set assay="RNA". 

```
# Run PCA for "RNA" assay; this is required for the JackStraw plot
sc = Seurat::RunPCA(sc, features=Seurat::VariableFeatures(object=sc), assay="RNA")

```

Reply:
Ignored for later.